### PR TITLE
Support GraphQL 4.0

### PR DESCRIPTION
**Pull request by esfomeado**
- Added support for GraphQL Plugin 4.0
- Moved to Java 17
- Fix some deprecations
- Cleanup

### CODE REVIEW

**Comment by srinivasankavitha** [comment]
Looks like you'll also need to update the build environment to use JDK 17 since tests are failing: https://github.com/Netflix/dgs-intellij-plugin/blob/main/.github/workflows/build.yml#L51

File: build.gradle.kts [hunk 1]
```diff
@@ -23,9 +23,9 @@ plugins {
     // Java support
     id("java")
     // Kotlin support
-    id("org.jetbrains.kotlin.jvm") version "1.8.10"
```
Comment (srinivasankavitha) [inline build.gradle.kts:26]: Can you change this back to 1.8.10? The build seems to fail locally as well otherwise.

Comment (esfomeado) [reply build.gradle.kts:26]: It worked fine when I tested but I downgraded it to 1.8.10

Comment (srinivasankavitha) [reply build.gradle.kts:26]: Build is green now, will merge and release

**Pull request merged**